packages/astro/src/content/utils.ts

```diff
@@ -68,8 +68,20 @@ export async function getEntrySlug(entry: Entry, collectionConfig: CollectionCon
 export async function getEntryData(entry: Entry, collectionConfig: CollectionConfig) {
 	let data = entry.data;
 	if (collectionConfig.schema) {
+		// TODO: remove for 2.0 stable release
+		if (
+			typeof collectionConfig.schema === 'object' &&
+			!('safeParseAsync' in collectionConfig.schema)
+		) {
```

Comment:
Seems like we could allow POJOs (plain ole' JS objects) here and automatically wrap them with `z.object` if necessary? Small convenience if you don't need any of the advanced `z.object` features.